Dockerfile

```diff
@@ -37,41 +17,68 @@ RUN apt-get update && \
+
+ENV VIRTUAL_ENV=/edx/app/edx_notes_api/venvs/edx_notes_api
 RUN python3.8 -m venv $VIRTUAL_ENV
 ENV PATH="$VIRTUAL_ENV/bin:$PATH"
 
 
+
 RUN locale-gen en_US.UTF-8
 ENV LANG en_US.UTF-8
 ENV LANGUAGE en_US:en
 ENV LC_ALL en_US.UTF-8
-ENV EDXNOTES_CONFIG_ROOT /edx/etc
-ENV DJANGO_SETTINGS_MODULE notesserver.settings.yaml_config
+
+
+# ENV variables lifetime is bound to the container whereas ARGS variables lifetime is bound to the image building process only
+# Also ARGS provide us an option of compatibility of Path structure for Tutor and other OpenedX installations
+ARG COMMON_CFG_DIR "/edx/etc"
+ARG COMMON_APP_DIR="/edx/app"
+ARG EDX_NOTES_API_APP_DIR="${COMMON_APP_DIR}/edx_notes_api"
+ARG EDX_NOTES_API_VENV_DIR="${COMMON_APP_DIR}/edx_notes_api/venvs/edx_notes_api"
+ARG EDX_NOTES_API_CODE_DIR="${INSIGHTS_APP_DIR}/edx_notes_api"
+
+ENV EDX_NOTES_API_APP_DIR ${EDX_NOTES_API_APP_DIR}
+
+# No need to activate edx_notes_api virtualenv as it is already activated by putting in the path
+# RUN virtualenv -p python3.8 --always-copy ${EDX_NOTES_API_VENV_DIR}
+
+COPY requirements ${EDX_NOTES_API_CODE_DIR}/requirements
+
+ENV PATH="${EDX_NOTES_API_CODE_DIR}/node_modules/.bin:$PATH"
+
+WORKDIR ${EDX_NOTES_API_CODE_DIR}/
+
+# edx_notes_api service config commands below
+RUN pip install  --no-cache-dir -r ${EDX_NOTES_API_CODE_DIR}/requirements/production.txt
+
+
+# Tried to cache the dependencies by copying related files after the npm install step but npm post install fails in that case.
+COPY . ${EDX_NOTES_API_CODE_DIR}/
+# RUN npm set progress=false && npm ci
 
 EXPOSE 8120
-RUN useradd -m --shell /bin/false app
+EXPOSE 18120
+
+
+FROM app as dev
 
-WORKDIR /edx/app/notes
+RUN pip install --no-cache-dir -r requirements/base.txt
 
-# Copy the requirements explicitly even though we copy everything below
-# this prevents the image cache from busting unless the dependencies have changed.
-COPY requirements/base.txt /edx/app/notes/requirements/base.txt
-COPY requirements/pip.txt /edx/app/notes/requirements/pip.txt
+ENV DJANGO_SETTINGS_MODULE "notesserver.settings.devstack"
 
-# Dependencies are installed as root so they cannot be modified by the application user.
-RUN pip install -r requirements/pip.txt
-RUN pip install -r requirements/base.txt
+# Backwards compatibility with devstack
+RUN touch "${EDX_NOTES_API_APP_DIR}/edx_notes_api_env" 
 
-RUN mkdir -p /edx/var/log
+CMD while true; do python ./manage.py runserver 0.0.0.0:8120; sleep 2; done
 
-# Code is owned by root so it cannot be modified by the application user.
-# So we copy it before changing users.
-USER app
+FROM app as prod
 
-# Gunicorn 19 does not log to stdout or stderr by default. Once we are past gunicorn 19, the logging to STDOUT need not be specified.
-CMD gunicorn --workers=2 --name notes -c /edx/app/notes/notesserver/docker_gunicorn_configuration.py --log-file - --max-requests=1000 notesserver.wsgi:application
+ENV DJANGO_SETTINGS_MODULE "notesserver.settings.devstack"
```

Comment:
so what should i replace with this line?
Since you are also using the same way I'm.
https://github.com/openedx/edx-analytics-dashboard/blob/master/Dockerfile

Reply:
Discussed with sobhan and updated production settings.